datafusion/common/src/functional_dependencies.rs

```diff
@@ -68,11 +68,26 @@ impl Constraints {
         let constraints = constraints
             .iter()
             .map(|c: &TableConstraint| match c {
-                TableConstraint::Unique {
-                    columns,
-                    is_primary,
-                    ..
-                } => {
+                TableConstraint::Unique { columns, .. } => {
+                    let field_names = df_schema.field_names();
+                    // Get primary key and/or unique indices in the schema:
+                    let indices = columns
+                        .iter()
+                        .map(|pk| {
+                            let idx = field_names
+                                .iter()
+                                .position(|item| *item == pk.value)
+                                .ok_or_else(|| {
+                                    DataFusionError::Execution(
+                                        "Unique key doesn't exist".to_string(),
```

Comment:
Can we include the field name in the error message to make it easier for the user to debug when they hit this error?

Reply:
as well as unique key details

Reply:
Sure thing, done :+1: